### DISCUSSION TITLE
How to create an observer that calculates total amount of drug that goes into a compartment

### DISCUSSION THREAD

**Discussion by akim2000** [Open-Systems-Pharmacology/Forum#1626] [Questions & Problems]
I am currently trying to create a brain model and need an observer to ensure that whatever changes I make to my model isn't somehow messing up the entire mass balance (i.e., I want observers to show me that: (total drug going into brain) = (total drug leaving brain + drug remaining in the brain + complexes formed)). But before looking at both sides, I wanted to focus on the first half, which is total drug going into the brain. 

From my knowledge, I'm pretty sure that all of the drug that reaches the brain (or any other organs) should be coming from the arterial blood compartment with the passive transport 'MassTransferBloodPool2OrgPl' (correct me if I am wrong). If so, I was wondering if it was possible to somehow observe and calculate the total amount of drug that is coming into the brain, so that I can compare with the second part of my equation (total drug leaving brain + drug remaining in the brain + complexes formed). How would I create an observer for this situation, or is there a better way to check that my mass balance isn't accidently messed up?

**Comment by PavelBal** [comment]
1. You should also consider the drug transported with blood cells, this would be the passive transport (PT) `MassTransferBloodPool2OrgRBC`.
2. By enabling the checkbox "Create process rate parameter", you can have access to the rate of the transport (you can do the same for reactions btw):

![image](https://github.com/Open-Systems-Pharmacology/Forum/assets/25903040/039c1955-9bcc-4831-8683-4daddde394d1)

3. Then you can create a "State variable parameter" (I think this does not work with observers), e.g. in the brain plasma (under `MoleculeProperties`), called e.g. `TotalAmountInflow`, mark it as "Parameter is state variable", and define its right hand side as the process rate of the inflow passive transport. Tha path to the process rate would be `Neighborhoods|ArterialBlood_pls_Brain_pls|MOLECULE|MassTransferBloodPool2OrgPl|ProcessRate`

![image](https://github.com/Open-Systems-Pharmacology/Forum/assets/25903040/d7acb197-01e4-4aa2-9d73-9e80ba3813f3)

**Comment by PavelBal** [comment]
Then do the same for inflow with the RBC, and for the outflows. Keep in mind, that if you are working with the model for large molecules, there is an additional outflow process via lymph and some endosomal degradation that you should consider in the mass balance.

**Comment by akim2000** [comment]
Where/how would I view the total drug inflow after creating the parameters?
